src/htm/engine/Network.cpp

```diff
@@ -44,6 +44,16 @@ Network::Network() {
   commonInit();
 }
 
+// move constructor
+Network::Network(Network && n) {
+  regions_ = std::move(n.regions_);
```

Comment:
Network(API) classes do not use the concept of `initialize()` method for Cereal to work? (needs implicit constructor + initialize() method)

Reply:
> (needs implicit constructor + initialize() method)

When Cereal restores it assumes that the original saved stream had been initialized before storing.  So, it tries to put everything back to where it was.

I found that if the original network had not been initialized before being serialized, the restore has a problem because some of the things it expected were not there.  I am looking into this and a few other serialization problems right now (to be in a follow-on PR).

The py/advanced unit tests have some non-trivial tests and it has exposed a few new problems I did not know we had. :-)

But to answer what I think is your question: 
- Before serialization we do constructor followed by adding regions and links, then initialize.  The initialize step will be performed by run( ) if needed.
- On restore we start with an empty constructor and then populate everything so we do not need the initialization step.